### DISCUSSION TITLE
Is DNN 9 Theme install folder wrong?

### DISCUSSION THREAD

**Discussion by thienvc** [dnnsoftware/Dnn.Platform#4886] [Q&A]
Is DNN 9 Theme install folder wrong?
After the installation, the Skin was thrown into the folder /Portals/[PortalID]-System/Skins
Why is that? I remember it should be in /Portals/[PortalID]/Skins?
I am using DNN 09.09.0

![image](https://user-images.githubusercontent.com/1115393/138805863-b3eb969e-521d-4bd6-a89f-519e56e8f268.png)

**Comment by valadas** [comment]
This is not expected behaviour.

although portal level themes are still technically supported, the UI should not allow doing it. I wonder if there is something special in the manifest of that theme ? Was it installed using the Persona Bar or some other control panel ?

**Reply by thienvc** [reply]
> This is not expected behaviour.
> 
> although portal level themes are still technically supported, the UI should not allow doing it. I wonder if there is something special in the manifest of that theme ? Was it installed using the Persona Bar or some other control panel ?

Normal installation operation through Personal Bar with the host account. Only the Host account (SuserUser) can install Themes, Admin account does not show Install button in both Themes menu and Extensions menu

I am trying to install theme https://github.com/2sic/dnn-theme-bootstrap4-instant/ of 2Sic @iJungleboy 
On DNN 09.09.0 and also 09.10.0

**I have tried removing and installing the Theme in 2 different ways, the result is:**
1. Install from PB menu `Settings => Extensions`: Theme has been installed to **`\Portals\_default\Skins\`**
2. Install from PB menu `Manage => Themes`: Theme has been installed to **`\Portals\0-System\Skins\`**

**Try uninstalling the theme:**
1. Remove the theme from PB menu `Manage => Themes`: Error message because the theme is installed as a Package, so it must be removed from Extensions
2. Remove the theme from PB menu `Settings => Extensions`: Only the theme located in **`\Portals\0-System\Skins\`** is removed, and the theme located in **`\Portals\_default\Skins\`** still remains.

**Comment by valadas** [comment]
The theme manifest looks about ok to me, that is strange. I am guessing it only happens on that specific instance and you cannot reproduce on a clean install ? It might need some digging around to figure out...

**Comment by bdukes** [comment]
The difference between `Portals/0` and `Portals/0-System` is that `Portal/0` could be offloaded to external storage (e.g. AWS, Azure, Dropbox, etc.). So site-specific themes and containers should be in the `-System` folder, since it always needs to be physically part of the file system.

Regarding the differences between the Themes and Extensions sections of the Persona Bar, I think it's probably not well defined what those expectations should be. The Themes section is visible to administrators whereas Extensions is only visible to super users, which is probably what leads to Themes always installing themes in a site-specific way. How it _should_ behave is an open question.

**Reply by thienvc** [reply]
Currently I see Administrator Account is not allowed to install Themes (in both from 2 places) And only non-packaged themes are allowed to be deleted.
So the site admin can only install the theme by uploading it normally via FTP or via File upload

**Comment by valadas** [comment]
Oh I see, so installing in Themes or in Extensions has different behaviour. I never even noticed you could install a theme from the Themes module 😄

**Comment by thienvc** [comment]
This is my first time trying to install a packaged Theme, before that I always upload the Theme by FTP, not package the theme.
However, I think this change is also very right when planning when Cache, Container, Theme into `Portals/0-System/`, this makes the `Portals/0/` directory cleaner and neater, will make File Browser faster and lighter.
DNN still supports the old standard backwards when the theme is in the `Portals/0/Skins` directory, but I think we should have clear documentation on this to avoid confusion for the community.

The question is if there are 2 themes with the same name, for example:
`/Portals/0/Skins/MyTheme/` and `/Portals/0-System/Skins/MyTheme/` which one should be used?
